natlib/svcd.c

```diff
@@ -183,5 +192,39 @@ static int svcd_init( lua_State *L )
         lua_call(L, 3, 0);
     }
 
+    
+    return 0;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// SVCD.ndispatch((string pay, number srcip, number srcport)) 
+// Authors: Jennifer Dai, Prashan Dharmasena, Wenqin 
+/////////////////////////////////////////////////////////////
+
+static int svcd_ndispatch( lua_State *L )
+{
+    // Checks to make sure it is passed 3 parameters
+    if (lua_gettop(L) != 3) return luaL_error(L, "Expected (pay, srcip, srcport)"); 
+
+    size_t parlen;
+    const char* pay = lua_tolstring(L, -1, &parlen); 
+    uint16_t ivkid = lua_tonumber(L, -1); // changes pay parameter to number and returns
+    lua_getglobal(L, "SVCD"); // gets table
+    lua_pushstring(L, "oursubs"); // use oursubs as the key
+    lua_gettable(L, 4);
+    char ivkidstr[16];
+    sprintf(ivkidstr, "%d", ivkid);
+    lua_pushstring(L, ivkidstr);
+    lua_gettable(L, 5);
```

Comment:
Why are you using the string version of the ivkid? I have to check what the other groups are doing when using the ivkid as a key, but the Lua version just straight up used the number, not the string. String conversion is slow and unnecessary.